services/reddit/user-karma.service.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { nonNegativeInteger } = require('../validators')
+const { metric } = require('../text-formatters')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  data: Joi.object({
+    link_karma: nonNegativeInteger,
+    comment_karma: nonNegativeInteger,
+  }).required(),
+}).required()
+
+module.exports = class RedditUserKarma extends BaseJsonService {
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'reddit/user-karma',
+      pattern: ':variant(link|comment|combined)/:user',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Reddit User Karma',
+        namedParams: { variant: 'combined', user: 'example' },
+        staticPreview: {
+          label: 'combined karma',
+          message: 56,
+          color: 'red',
+          style: 'social',
+        },
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'reddit karma',
+      namedLogo: 'reddit',
+    }
+  }
+
+  static render({ label, karma, user }) {
+    return {
+      label: `${label}`,
+      message: metric(karma),
+      color: 'red',
+      link: [`https://www.reddit.com/u/${user}`],
+    }
+  }
+
+  async fetch({ user }) {
+    return this._requestJson({
+      schema,
+      url: `https://www.reddit.com/u/${user}/about.json`,
+      errorMessages: {
+        404: 'user not found',
+      },
+    })
+  }
+
+  async handle({ variant, user }) {
+    const json = await this.fetch({ user })
+
+    let karma
+    let label
+    if (variant === 'link') {
+      karma = json.data.link_karma
+      label = `${user}'s karma (link)`
+    } else if (variant === 'comment') {
+      karma = json.data.comment_karma
+      label = `${user}'s karma (comment)`
+    } else {
+      const total = json.data.link_karma + json.data.comment_karma
+      karma = total
+      label = `${user}'s karma`
```

Comment:
The role for constructing the `label` value is a rendering related concern so it should reside in the `render` function. I'm also not sure how grammatically safely we can (or should) attempt to pluralize the `user` (i.e. avoiding `foos's karam`)

Reply:
The main reason I kept the construction of `label` in the handle function was to prevent needlessly running the same if statement multiple times for a single badge, but can move it into the render function if that's preferred. 

I see what you mean about pluralising the `user`, and again can remove it if that the preferred shields option, I just felt it would be much more natural being plural. For example with my username, `JoeIzzard's Karma` seems more natural, especially for a social badge, than `JoeIzzard Karma`. I think either way we are assuming something so it's a case of what fits with shields better from and standard style point?

Reply:
> The main reason I kept the construction of label in the handle function was to prevent needlessly running the same if statement multiple times for a single badge, but can move it into the render function if that's preferred

I suspected that was the case. However, the benefits of the separation of concerns outweigh any minor performance gains and will be more consistent with the Shields style/approach. Also, the this can be expressed succinctly within `render` by including `variant` is a param and only needing one conditional expression evaluation.

Something like
```js
const label = variant === 'combined' ? `${user}'s karma` : `${user}'s karma (${variant})`
```

Reply:
> I just felt it would be much more natural being plural

Is there any kind of symbol prefix/suffix that could be used? I'm only an occasional and brief browser of reddit so not terribly familiar with the platform. For example `r/User karma` or even something like what we do for the Twitter badges `@User karma` (understood the `@` has a specific meaning on that platform that may not apply to Reddit)

Reply:
That's a good idea, I will go that route (Reddit uses /u/username for users). Will also move the label construction to render, and fix a hard coded username in the handle function that shouldn't be there either.